src/storage/src/source/kafka.rs

```diff
@@ -564,6 +565,26 @@ impl SourceRender for KafkaSourceConnection {
                 // the consumer directly.
                 while let Some(result) = reader.consumer.poll(Duration::from_secs(0)) {
                     match result {
+                        Ok(message) => {
+                            let (message, ts) =
+                                construct_source_message(&message, &reader.metadata_columns);
+                            if let Some((msg, time, diff)) = reader.handle_message(message, ts) {
+                                let pid = time.partition().unwrap();
+                                let part_cap = &reader.partition_capabilities[pid].data;
+                                match msg {
+                                    Ok(msg) => {
+                                        data_output.give(part_cap, ((0, Ok(msg)), time, diff)).await
+                                    }
+                                    Err(err) => {
+                                        let err = SourceReaderError::other_definite(anyhow!(err));
+
+                                        data_output
+                                            .give(part_cap, ((0, Err(err)), time, diff))
+                                            .await
+                                    }
+                                }
```

Comment:
You can factor the error mapping out to reuse the rest of the code. 


```suggestion
                                let msg = msg.map_err(|e| SourceReaderError::other_definite(anyhow!(e)));
                                data_output.give(part_cap, ((0, msg), time, diff)).await;
```

Also, it would be nice if `DecodeError` implemented `Error` from the stdlib so that you can just do `e.into()` instead of wrapping it in `anyhow!(..)`. This looks like an omission from the original definition. Totally optional for this PR, feel free to ignore this last part.